### PR TITLE
job runner: limit concurrency

### LINKED ISSUE #2680: fileinstall: restore parallism limiting
We used to manage file installation processes (e.g. `rsync`s) in a worker pool. We switched to an asyncio implementation, however, the limit was lost in the process:

https://github.com/metomi/rose/pull/2325/files#diff-1a4d9963cf73c867e4dcfdce40251652a0674c8220771f24138b54ef9d4eaf3cL152

We should restore this parallel limit to reduce the risk of issues which could result from it.

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Restore the Rose 2019 concurrency limit of 6.
* Closes #2680

Had to jig the code around a bit to move away from the maximum concurrency model.

For review, please ensure:

* It limits concurrency.
* It still handles errors correctly.
* It runs all the jobs (I think some jobs can be inter-dependent?).

TODO: Need to figure out which branch to put this on => Monday problem.

### CODE REVIEW

File: metomi/rose/job_runner.py [hunk 1]
```diff
@@ -161,87 +163,96 @@ def __init__(self, job_processor, nproc=None):
...
-                # jobs have become ready and submit those.
-                just_completed, awaiting = loop.run_until_complete(
-                    asyncio.wait(awaiting, return_when=asyncio.FIRST_COMPLETED)
+            while len(running) < concurrency:
+                # run jobs
+                job = job_manager.get_job()
+                if job is None:
+                    # we've run out of jobs for now
+                    break
+                task = asyncio.create_task(
```
Comment (oliver-sanders) [inline metomi/rose/job_runner.py:214]: To test the concurrency jam a print statement in here.

**Comment by oliver-sanders** [comment]
> TODO: Need to figure out which branch to put this on => Monday problem.

Monday answer, this needed to go into "2.0.x".

**Comment by MetRonnie** [comment]
I hope it's the PR that is posthumous rather than me 😄

**Comment by oliver-sanders** [comment]
Ok, cherry-picked and re-raised - https://github.com/metomi/rose/pull/2684

**Pull request merged**